jax/lax/lax.py

```diff
@@ -4612,12 +4682,24 @@ def _common_reduce_window_shape_rule(operand, window_dimensions,
     msg = ("reduce_window got inconsistent window_strides and "
            "window_dimensions: got window_strides {} and window_dimensions {}.")
     raise TypeError(msg.format(window_strides, window_dimensions))
+  if len(base_dilation) != len(window_dimensions):
+    msg = ("reduce_window got inconsistent base_dilation and "
+           "window_dimensions: got base_dilation {} and window_dimensions {}.")
+    raise TypeError(msg.format(base_dilation, window_dimensions))
+  if len(window_dilation) != len(window_dimensions):
+    msg = ("reduce_window got inconsistent window_dilation and "
+           "window_dimensions: got window_dilation {} and window_dimensions "
+           "{}.")
+    raise TypeError(msg.format(window_dilation, window_dimensions))
 
   return reduce_window_shape_tuple(operand.shape, window_dimensions,
-                                   window_strides, padding)
+                                   window_strides, padding, base_dilation,
+                                   window_dilation)
 
 def reduce_window_shape_tuple(operand_shape, window_dimensions, window_strides,
-                              padding):
+                              padding, base_dilation, window_dilation):
```

Comment:
If we had defaults for `base_dilation` and `window_dilation` here, we could avoid having to make downstream changes in the two consumers of `reduce_window_shape_tuple` (neural tangents and oryx). But default args are also bad, so maybe it's not worth it?

Reply:
True.

Actually what we should do here is convince Neural Tangents and Oryx to use `jax.eval_shape` instead, rather than this undocumented API...?

I'll add the defaults for now.

Reply:
Filed https://github.com/google/neural-tangents/issues/51 and emailed for Oryx.

Reply:
Looks like it's also used in Flax; I filed https://github.com/google/flax/issues/370.